corehq/apps/hqwebapp/static/hqwebapp/js/inactivity.js

```diff
@@ -74,7 +75,12 @@ hqDefine('hqwebapp/js/inactivity', [
             log("Could not find modal, returning");
             return;
         }
-
+        if ($newVersionModal.length) {
+            $('#refreshApp').click(function () {
+                history.pushState("", document.title, window.location.pathname + window.location.search);
```

Comment:
What does this do? Isn't the current state already in the history? I thought we 403ed if you went to web apps with an app id that isn't the current version, which would mean that going back after the reload wouldn't work anyway.

Reply:
When we 403, we show the user a message saying "Permission Denied" Which I felt is not a good User Experience given that we are ourselves reloading the page. However, if we redirect them to the Apps page directly then the error message is not shown.

Reply:
Definitely agreed about the UX and not 403ing. So is this setting up the history so that when you `reload` it goes to the apps page instead of attempting to refresh whatever screen they were on?

Reply:
Yeah. That is correct.

Reply:
Makes sense, though I think it'd be slightly clearer use a `document.location = ...`, since it feels more like a redirect than a refresh.

Reply:
Done in https://github.com/dimagi/commcare-hq/pull/30778/commits/2a5aae36390314e28d97a93e8d6f4f2e4864ea1a